src/Modules/OrchardCore.Commerce/Endpoints/Services/ShoppingCartService.cs

```diff
@@ -77,11 +76,11 @@ public async Task<string> AddItemAsync(ShoppingCartLineUpdateModel line, string
         }
         catch (FrontendException ex)
         {
-            var errors = ex.HtmlMessages;
-            errored = errors.ConvertLocalizedHtmlStringList();
+            var errors = ex.HtmlMessages.Select(error => error.Html());
+            strError = string.Join(System.Environment.NewLine, errors);
         }
 
-        return errored;
+        return strError;
```

Comment:
Please use `try { return ... }` above, then you can remove this line because all execution paths will have returned by then.